File: sdks/android/src/main/kotlin/com/apexmediation/sdk/ApexMediation.kt
package com.apexmediation.sdk

import android.content.Context
import android.os.Handler
import android.os.Looper

object ApexMediation {
    private var initialized = false

    fun initialize(context: Context, apiKey: String, callback: (Result<Unit>) -> Unit) {
        if (apiKey.isBlank()) {
            callback(Result.failure(IllegalArgumentException("Invalid API key")))
            return
        }

        Handler(Looper.getMainLooper()).postDelayed({
            initialized = true
            callback(Result.success(Unit))
        }, 250)
    }

    fun requestInterstitial(placementId: String, callback: (Result<AdFill>) -> Unit) {
        if (!initialized) {
            callback(Result.failure(IllegalStateException("SDK not initialized")))
            return
        }

        if (placementId.isBlank()) {
            callback(Result.failure(IllegalArgumentException("Invalid placement")))
            return
        }

        val fill = AdFill(
            adapter = "admob",
            ecpm = 12.3,
            creativeUrl = "https://ads.apexmediation.com/interstitial.mp4"
        )
        callback(Result.success(fill))
    }
}

data class AdFill(
    val adapter: String,
    val ecpm: Double,
    val creativeUrl: String
)

File: sdks/unity/Runtime/ApexMediation.cs
using System;
using System.Collections;
using UnityEngine;

namespace ApexMediation
{
    public class Mediation : MonoBehaviour
    {
        private static Mediation _instance;
        private bool _initialized;

        public static Mediation Instance
        {
            get
            {
                if (_instance == null)
                {
                    var go = new GameObject("ApexMediation");
                    DontDestroyOnLoad(go);
                    _instance = go.AddComponent<Mediation>();
                }

                return _instance;
            }
        }

        public void Initialize(string apiKey, Action<bool> callback)
        {
            if (string.IsNullOrEmpty(apiKey))
            {
                callback?.Invoke(false);
                return;
            }

            StartCoroutine(InitializeCoroutine(callback));
        }

        private IEnumerator InitializeCoroutine(Action<bool> callback)
        {
            yield return new WaitForSeconds(0.25f);
            _initialized = true;
            callback?.Invoke(true);
        }

        public void RequestInterstitial(string placementId, Action<AdFill> callback)
        {
            if (!_initialized)
            {
                callback?.Invoke(null);
                return;
            }

            if (string.IsNullOrEmpty(placementId))
            {
                callback?.Invoke(null);
                return;
            }

            var fill = new AdFill
            {
                Adapter = "admob",
                Ecpm = 12.3,
                CreativeUrl = "https://ads.apexmediation.com/interstitial.mp4"
            };

            callback?.Invoke(fill);
        }
    }

    [Serializable]
    public class AdFill
    {
        public string Adapter;
        public double Ecpm;
        public string CreativeUrl;
    }
}
